src/editor/utils/index.js

```diff
@@ -149,19 +150,48 @@ export const deepCopy = object => {
   return obj
 }
 
-export const loadImage = url => {
-  const image = new Image()
+export const loadImage = (url, detectContentType) => {
   return new Promise((resolve, reject) => {
-    image.onload = () => {
-      resolve(url)
-    }
-    image.onerror = err => {
+    try {
+      if (detectContentType && !checkImageContentType(url)) {
```

Comment:
checkImageContentType(url) will return a promise, a promise is a `object`, So `! checkImageContentType` is always be `false`.

You should not put a promise in `if conditions`.

```javascript
const loadImage = async (url, detectContentType) => {
  const isImage = await checkImageContentType(url)
  if (detectContentType && isImage) {
    // other codes
  }
  // other codes
}
```

Reply:
I updated PR. Do you want merge this or should we fix "http request" issue first? If you want to merge, I have to remove logging.